tensorflow_datasets/core/dataset_builder_test.py

```diff
@@ -324,6 +324,18 @@ def test_non_preparable_version(self, *unused_mocks):
     self.assertIsNotNone(builder)
     with self.assertRaisesWithPredicateMatch(AssertionError, expected):
       builder.download_and_prepare()
+      
+  def test_with_old_versions(self):
+    expected = (
+        "The version of the dataset you are trying to use"
+        " (dummy_dataset_shared_generator:0.0.8) cannot be"
+        " generated as this version is too older to use."
+        " Please  use another new availble version of the dataset"
+        " (new availble version: ['1.0.0', '2.0.0'])")
+    builder = DummyDatasetSharedGenerator(version="0.0.8")
```

Comment:
I think it's best to create a new specific dataset in this file for this.
This allow the test to be self-contained and reader don't have to look into multiple files to understand the test..

Reply:
Thanks for review this, I will make changes soon

Reply:
By the way, why haven't you done this ? As explain it's best for datasets to be self-contained.